packages/flutter/test/material/buttons_test.dart

```diff
@@ -1191,4 +1192,66 @@ void main() {
     final Material material = tester.widget<Material>(rawButtonMaterial);
     expect(material.color, const Color(0xff00ff00));
   });
+
+  testWidgets('MaterialButton should be enabled when onLongPress is not null.', (WidgetTester tester) async {
+
+    final Finder rawButtonMaterial = find.descendant(
+      of: find.byType(MaterialButton),
+      matching: find.byType(Material),
+    );
+
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: MaterialButton(
+          color: const Color(0xffff00ff),
+          onPressed: null,
+          onLongPress: () { /* to make sure the button is enabled */ },
+          child: const Text('button'),
+        ),
+      ),
+    );
+
+    final Material material = tester.widget<Material>(rawButtonMaterial);
+    expect(material.color, const Color(0xffff00ff));
```

Comment:
I don't think checking `material.color` is sufficient in proving that the button is enabled. I think it is more precise to check if the button's enabled property is true/false.

Reply:
Fixed on last commits. And added tests for the other buttons.